memgpt/server/server.py

```diff
@@ -1292,3 +1295,36 @@ def detach_source_from_agent(self, user_id: uuid.UUID, agent_id: uuid.UUID, sour
     def list_attached_sources(self, agent_id: uuid.UUID):
         # list all attached sources to an agent
         return self.ms.list_attached_sources(agent_id)
+
+    def list_data_source_passages(self, user_id: uuid.UUID, source_id: uuid.UUID) -> List[PassageModel]:
+        warnings.warn("list_data_source_passages is not yet implemented, returning empty list.", category=UserWarning)
```

Comment:
@sarahwooders I implemented these as stub methods for now so the metadata will always return 0 passages, 0 documents, we can move implementation to a separate future PR